wpengine/faustjs
# MERL-1270: Do not rewrite links in the post editor

## Tasks

- [x] phpunit tests added

## Description
When Faust’s “Enable Post and Category URL rewrites” is enabled, this has been causing URLs to be rewritten in the editor to contain the front-end URL instead of the WP URL, which causes them to be saved to the database this way.

Related: the `guid` value for the post is also saved with the Faust front-end URL. This should also be saved with the WP domain.

Ideally the WP URLs are saved to the database, so that:
- The content is consistent
- Our rewrite logic can properly target all desired URLs

## Steps to Reproduce
- Install and activate the Faust WP plugin
- Add a front-end URL to the Faust settings. Leave “Enable Post and Category URL rewrites” enabled (which is the default)
- Create a new post. Add a text link to another post. Note the front-end URL shows in the search results. Save the post.
- Note the front-end URL was saved in `post_content` and `guid` instead of the WP URL.

## Expected Behavior
The WP URL is saved to the database, so that the content is consistent and matches the domain/URL where it actually lives.

## Testing
- Check out this branch
- Create a new post, follow the steps above.
- Note the links use the WP domain and are saved in `post_content` and `guid`.
- Try to break it! Look for flaws in the logic. 

## Documentation Changes



## Dependant PRs
